pkg/client/client.go

```diff
@@ -445,11 +456,16 @@ func (c *SPDKClient) EngineSnapshotCreate(name, snapshotName string, opts *api.S
 	ctx, cancel := context.WithTimeout(context.Background(), GRPCServiceTimeout)
 	defer cancel()
 
-	resp, err := client.EngineSnapshotCreate(ctx, &spdkrpc.SnapshotRequest{
+	snapshotRequest := spdkrpc.SnapshotRequest{
 		Name:         name,
 		SnapshotName: snapshotName,
 		UserCreated:  opts != nil && opts.UserCreated,
-	})
+	}
+	if opts != nil {
+		snapshotRequest.SnaphotTimestamp = opts.Timestamp
```

Comment:
Same as above

Reply:
Here `SnapshotTimestamp` is not needed, I removed it. Actually, not even `UserCreated` is needed, so the `opts` parameter is unnecessary, maybe it is better to remove it after this PR will be merged. WDYT?

Reply:
> not even UserCreated is needed

Can you elaborate more on it?

Reply:
Both user created and snapshot timestamp flags are internally resolved in `longhorn-spdk-engine`, so it is not needed to insert them as parameters in the function `EngineSnapshotCreate`, which is called by `longhorn-instance-manager`.

Reply:
I see. Then, we can remove the parameter `opts`.

Reply:
Why not remove it in this PR?